QuickNoteVault.API/QuickNoteVault.DAL/UOW/UnitOfWork.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage;
+using QuickNoteVault.DAL.Repository;
+
+namespace QuickNoteVault.DAL.UOW
+{
+    public class UnitOfWork : IUnitOfWork
+    {
+        private readonly DbContext _context;
+        private readonly IDictionary<Type, object> _repositories = new Dictionary<Type, object>();
+        private bool _disposed = false;
+
+        public UnitOfWork(DbContext context)
+        {
+            _context = context;
+        }
+
+        public IRepository<T> GetRepository<T>()
+            where T : class
+        {
+            var type = typeof(T);
+            if (!_repositories.ContainsKey(type))
```

Comment:
Use `TryGetValue` instead of `ContainsKey`

`if (!_repositories.TryGetValue(type, out object? repo))
{
    repo = new Repository<T>(_context);
    _repositories.Add(type, repo);
}`

Add space between `var type` and `if` statement